node/lib/util/repo_status.js

```diff
@@ -121,146 +121,14 @@ class Submodule {
-        // been added.
-
-        if (FILESTATUS.ADDED === this.d_indexStatus) {
-            assert.isNull(this.d_commitSha);
-            assert.isNull(this.d_commitUrl);
-        }
-        else {
-            assert.isNotNull(this.d_commitSha);
-            assert.isNotNull(this.d_commitUrl);
-        }
-
-        // If there is a modification, then one of `indexSha` or `indexUrl`
-        // must differ from the commit.
-
-        if (FILESTATUS.MODIFIED === this.d_indexSha) {
-            assert(this.d_commitSha !== this.d_indexSha ||
-                   this.d_commitUrl !== this.d_indexUrl);
-        }
-
-        // No modification is indicated, but index and commit differ.
-
-        if (null === this.d_indexStatus) {
-            assert.equal(this.d_indexSha, this.d_commitSha);
-            assert.equal(this.d_indexUrl, this.d_commitUrl);
-        }
-
-        // Check to see that sha relations make sense.
-
-        // First, validate that when we have both commit and index shas, we
-        // also have a relation, and that the relation does not conflict with
-        // the values of the shas.
-
-        if (null !== this.d_indexSha && null !== this.d_commitSha) {
-            assert.isNotNull(this.d_indexShaRelation);
-            if (this.d_indexSha === this.d_commitSha) {
-                assert.equal(this.d_indexShaRelation, COMMIT_RELATION.SAME);
-            }
-            else {
-                assert.notEqual(this.d_indexShaRelation, COMMIT_RELATION.SAME);
-            }
-        }
-        else {
-            assert.isNull(this.d_indexShaRelation);
-        }
-
-        // First, validate that when we have both an index sha and a repo
-        // status with a head commit, we also have a relation, and that the
-        // relation does not conflict with the values of the shas.
-
-        if (null !== this.d_repoStatus &&
-            null !== this.d_repoStatus.headCommit &&
-            null !== this.d_indexSha) {
-            assert.isNotNull(this.d_workdirShaRelation);
-            if (this.d_repoStatus.headCommit === this.d_indexSha) {
-                assert.equal(this.d_workdirShaRelation, COMMIT_RELATION.SAME);
-            }
-            else {
-                assert.notEqual(this.d_workdirShaRelation,
-                                COMMIT_RELATION.SAME);
-            }
-        }
-        else {
-            assert.isNull(this.d_workdirShaRelation);
-        }
-
-
+        this.d_indexStatus        = status.indexStatus || null;
+        this.d_indexSha           = status.indexSha || null;
+        this.d_indexShaRelation   = status.indexShaRelation || null;
+        this.d_indexUrl           = status.indexUrl || null;
+        this.d_commitSha          = status.commitSha || null;
+        this.d_commitUrl          = status.commitUrl || null;
+        this.d_workdirShaRelation = status.workdirShaRelation || null;
+        this.d_repoStatus         = status.repoStatus || null;
         Object.freeze(this);
```

Comment:
Could we maybe leave the version with assertions under an if (debug) {} or something?